scaleway/loadbalancers.go

```diff
@@ -1706,3 +1792,68 @@ func getHTTPSHealthCheck(service *v1.Service, nodePort int32) (*scwlb.HealthChec
 		URI:    uri,
 	}, nil
 }
+
+func parsePrivateNetworkAnnotations(service *v1.Service) []privateNetworkAnnotations {
+	rawPrivatesNetworksAnnotations := service.Annotations[serviceAnnotationLoadBalancerPrivateNetworks]
+	var privateNetworksAnnotations []privateNetworkAnnotations
+
+	if len(rawPrivatesNetworksAnnotations) == 0 {
+		return privateNetworksAnnotations
+	}
+
+	for _, privateNetwork := range strings.Split(rawPrivatesNetworksAnnotations, ";") {
+
+		split := strings.Split(privateNetwork, ":")
+
+		if len(split) == 1 {
+			// DCHP
+			privateNetworksAnnotations = append(privateNetworksAnnotations, privateNetworkAnnotations{
+				privateNetworkID: split[0],
+				DHCPConfig:       &scwlb.PrivateNetworkDHCPConfig{},
+			})
+		} else {
+			// static ip addresse
+			ipAdresses := strings.Split(split[1], ",")
+			privateNetworksAnnotations = append(privateNetworksAnnotations, privateNetworkAnnotations{
+				privateNetworkID: split[0],
+				StaticConfig: &scwlb.PrivateNetworkStaticConfig{
+					IPAddress: ipAdresses,
+				},
+			})
+		}
+		return privateNetworksAnnotations
+
+	}
+	return privateNetworksAnnotations
+}
+
+type privateNetworkAnnotations struct {
+	privateNetworkID string
+	DHCPConfig       *scwlb.PrivateNetworkDHCPConfig
+	StaticConfig     *scwlb.PrivateNetworkStaticConfig
+}
+
+func (privateNetworkAnnotations privateNetworkAnnotations) privateNetworkExist(privateNetworks []*scwlb.PrivateNetwork) bool {
+	for _, privateNetwork := range privateNetworks {
+		if privateNetwork.PrivateNetworkID == privateNetworkAnnotations.privateNetworkID {
+			return true
+		}
+	}
+	return false
+}
+
+func privateNetworkAnnotationsIsPresentAndUpToDate(privateNetworksAnnotations []privateNetworkAnnotations, privateNetwork scwlb.PrivateNetwork) bool {
+	for _, privateNetworkAnnotations := range privateNetworksAnnotations {
+		// Check if
+		//   * id match
+		//   * DHCPConfig are equal (nil == nil or emptyObj == emptyOBJ)
+		//   * StaticConfig are equal (nil == nil or IPAddress == IPAddress)
+		if privateNetworkAnnotations.privateNetworkID == privateNetwork.PrivateNetworkID &&
+			privateNetworkAnnotations.DHCPConfig == privateNetwork.DHCPConfig &&
+			(privateNetworkAnnotations.StaticConfig == privateNetwork.StaticConfig ||
+				reflect.DeepEqual(privateNetworkAnnotations.StaticConfig.IPAddress, privateNetwork.StaticConfig.IPAddress)) {
```

Comment:
This will only works if the order is the same in both arrays, better loop on them I guess